Issue/2040 chart negative revenue design fixes

anitaa1990 opened the pull request:
Fixes #2027 by adding some design fixes fo supporting negative revenue stats.

#### Changes
- The y-axis labels should display only the min revenue, 0 and max revenue values.
- The bar chart corners should be rounded

#### Screenshots
<img width="300" src="https://user-images.githubusercontent.com/22608780/76715843-0b235480-6754-11ea-8da3-e5870cc854d0.png">. <img width="300" src="https://user-images.githubusercontent.com/22608780/76715845-0d85ae80-6754-11ea-84f3-71bfc93a09ce.png">. <img width="300" src="https://user-images.githubusercontent.com/22608780/76715846-0e1e4500-6754-11ea-9538-6380ff3ce47c.png">. <img width="300" src="https://user-images.githubusercontent.com/22608780/76715847-0eb6db80-6754-11ea-9dda-7d5290835e9e.png">
 
##### Old Stats UI
<img width="300" src="https://user-images.githubusercontent.com/22608780/76716290-b5e84280-6755-11ea-968d-aa6e9e4c0e28.png">. <img width="300" src="https://user-images.githubusercontent.com/22608780/76716293-b7b20600-6755-11ea-9f98-948335b533d4.png">. <img width="300" src="https://user-images.githubusercontent.com/22608780/76716294-b8e33300-6755-11ea-9dbb-b30678fa79d1.png">. <img width="300" src="https://user-images.githubusercontent.com/22608780/76716295-b97bc980-6755-11ea-806f-6647a6acd8d9.png">

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

anitaa1990 committed:
Added logic to display only the min, 0 and max revenue values to charts

anitaa1990 committed:
Added logic to display rounded corners to both old and new stats charts

Garance91540 reviewed (commented):
Hi @anitaa1990  Thank you so much for your work on this, it's definitely been a trickier exercise than I thought! 

- Rounded corners: The data bars look great, it's definitely clearer to see where they start/end with this formatting so thanks! 

- "The y-axis labels should display only the min revenue, 0 and max revenue values."
Looking at your screenshots and what happens if we have a high positive revenue and a small negative revenue, it doesn't seem that what I suggested was right: if we have $5000 in positive revenue and -$50 in negative revenue for example, the graph looks really weird by going from -$5000 to $5000. Would it be possible to revert it to what you had with the 3 "normal" data axis + a 4th "floating" zero axis (but making the zero axis a dotted line?) https://d.pr/i/xDtxtv

I don't think this is the final solution but I need more time to craft a sound alternative that works for all scenarios. I apologize for the extra work !

anitaa1990 commented:
Thanks for the review @Garance91540!

> The y-axis labels should display only the min revenue, 0 and max revenue values."
Looking at your screenshots and what happens if we have a high positive revenue and a small negative revenue, it doesn't seem that what I suggested was right: if we have $5000 in positive revenue and -$50 in negative revenue for example, the graph looks really weird by going from -$5000 to $5000. Would it be possible to revert it to what you had with the 3 "normal" data axis + a 4th "floating" zero axis (but making the zero axis a dotted line?)

I've reverted the change and updated the screenshots below (except the dotted line for the zeroline). Let me know if this is fine.

<img width="300" src="https://user-images.githubusercontent.com/22608780/76823557-9840d900-683a-11ea-9706-5643680d4380.png">. <img width="300" src="https://user-images.githubusercontent.com/22608780/76823562-9aa33300-683a-11ea-9ea5-5bab3c2f48e1.png">. <img width="300" src="https://user-images.githubusercontent.com/22608780/76823563-9b3bc980-683a-11ea-926c-46c86d9fc2a4.png">. <img width="300" src="https://user-images.githubusercontent.com/22608780/76823564-9bd46000-683a-11ea-912f-2a8c199ba414.png">

##### Old stats UI
<img width="300" src="https://user-images.githubusercontent.com/22608780/76823644-d2aa7600-683a-11ea-897a-1240631cdb1e.png">. <img width="300" src="https://user-images.githubusercontent.com/22608780/76823649-d5a56680-683a-11ea-985f-7e7ac8aa5a77.png">. <img width="300" src="https://user-images.githubusercontent.com/22608780/76823652-d76f2a00-683a-11ea-80e3-96e33aa7a80c.png">. <img width="300" src="https://user-images.githubusercontent.com/22608780/76823654-d8a05700-683a-11ea-8b52-ac94991d2be5.png">

I was looking into adding a dashed line for the zero axis and it looks like this is currently not supported in the MpAndroidChart (the library we use for displaying the charts). We are going to have to come up with a custom solution for this use case and I prefer to tackle this in a separate PR, if that is fine with you (already opened a separate issue [here](https://github.com/woocommerce/woocommerce-android/issues/2040)).

anitaa1990 committed:
Reverted logic to display only min value, 0 or max value in y -axis

Garance91540 reviewed (dismissed):
Looking great, thanks for the changes and no problem with opening a separate PR for the zero axis!

nbradbury commented:
@anitaa1990 I have zero stats but the bar chart shows otherwise. This problem goes away when I switch to `develop`.

![Screenshot_1584455042](https://user-images.githubusercontent.com/3903757/76866145-151f8300-683a-11ea-869e-a6993a2b0531.png)

anitaa1990 committed:
Removed test data accidentally getting committed

anitaa1990 commented:
> I have zero stats but the bar chart shows otherwise. This problem goes away when I switch to develop.

😱I made the colossal mistake of committing test data to the `MyStoreStatsView` 🤦‍♀! Reverted that in    e8f9797. My bad!!

nbradbury reviewed (approved):
:shipit:
